File: backend/src/main/java/org/pupille/backend/mysql/survey/umfrage/UmfrageDTO.java
package org.pupille.backend.mysql.survey.umfrage;

import lombok.Data;
import org.pupille.backend.mysql.survey.auswahloption.AuswahloptionNestedDTO;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

@Data
public class UmfrageDTO {
//    private Long unr;
    private UUID unr;
    private String anlass;
    private LocalDate endDatum;
    private String beschreibung;
    // We only nest the DTO (namely AuswahloptionDTO), not the Entity (namely Auswahloption)
    private List<AuswahloptionNestedDTO> auswahloptionendtos;
}

File: frontend/src/types/UmfrageDTO.ts
// UmfrageDTO
import {AuswahloptionNestedDTO} from "./AuswahloptionNestedDTO.ts";

export interface UmfrageDTO {
    unr: string;       // ID of the survey (null/undefined for new ones)
    anlass: string;
    endDatum: string; // ISO Date String 'YYYY-MM-DD'
    beschreibung: string;
    auswahloptionendtos: AuswahloptionNestedDTO[];
}
